### DISCUSSION TITLE
Is there any special configuration when running the swarm deploy behind a reverse proxy?

### DISCUSSION THREAD

**Discussion by andricDu** [BlueBrain/nexus#2831] [Q&A] [answered]
Hi,

We are trying to deploy BBN with docker swarm. We have a FQDN with certs setup for our reverse proxy which then forwards to the swarm. 

Everything seems to work correct, users can login with keycloak, UI loads, we can list the orgs and projects, however we get 404s on any paths related to `/resources/org/project/schema/id` https://github.com/BlueBrain/nexus/blob/master/delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala#L154

As I mentioned, everything else appears to work, including project creation. Even more strangely, when forgoing the domain and hitting it via IP directly, it appears to work fine.

The 404 appears to be returned by delta, and not nginx, but there are no logs within it to inspect. 

I was wondering if there is some config value we missed for proxy awareness.

**Comment by bogdanromanx** [comment]
Hi @andricDu,

The only configuration that needs to be done when running behind a proxy is to set the following property:
```
-Dapp.http.base-uri=http://delta:8080/v1
```
... with a value that matches your domain and path prefix.

Could you please check that you're running version 1.5.1 of Delta? I remember an issue in 1.5.0 related to this...

**Reply by andricDu** [reply]
I believe it is 1.5.1 but I will verify. 

I have a new suspicion that it is our nginx reverse proxy. The `{id}` component shows as a url encoded URI, but I believe the proxy pass functionality of nginx is mangling the the url encoding. I will let you know if this is definitely the case one way or the other.

**Comment by andricDu** [comment, marked as answer]
Turns out there was an Apache reverse proxy in front of our deployment. 
Due to the structure of ids for the schemas, we needed the following settings applied to our reverse proxy:
`AllowEncodedSlashes NoDecode` & `nocanon`

**Reply by bogdanromanx** [reply]
Happy that you managed to sort this out and post back.
